### PR TITLE
Err codes

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @jonathandturner

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -1527,6 +1527,37 @@ fn main() {
 ```
 "##,
 
+E0478: r##"
+A lifetime bound was not satisfied.
+
+Erroneous code example:
+
+// Check that the explicit lifetime bound (`'b`, in this example) must
+// outlive all the superbound from the trait (`'a`, in this example).
```
Comment (jonas-schievink) [inline src/librustc/diagnostics.rs]: They aren't named `'b` and `'a` though

**Comment by jonathandturner** [comment]
@bors r-

@GuillaumeGomez - will let you work on @jonas-schievink's comments

File: src/librustc/diagnostics.rs (hunk 1 above)
Comment (jonas-schievink) [inline src/librustc/diagnostics.rs]: Also "superbounds"

File: src/librustc/diagnostics.rs [hunk 2]
```diff
@@ -1527,6 +1527,37 @@ fn main() {
...
+
+In this example, the `'SnowWhite` lifetime is supposed to outlive the `'kiss`
+lifetime but the declaration of the `Prince` struct doesn't enforce it. To fix
+this issue, you need to specify it:
+
+```
+trait Wedding<'t>: 't { }
+
+struct Prince<'kiss, 'SnowWhite: 'kiss> { // You say here that 'b must live
+                                          // longer than 'a.
```
Comment (jonas-schievink) [inline src/librustc/diagnostics.rs]: Here too

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**